### PR TITLE
Add projects api

### PR DESCRIPTION
**Pull request by LMBishop**
Solves 13, and partially 17

### CODE REVIEW

File: src/main/java/org/kainos/ea/api/ProjectService.java [hunk 1]
```diff
@@ -21,4 +28,36 @@ public Project getProjectById(int id) throws FailedToGetProjectException, Projec
 
         return project;
     }
+
+    public int createProject(ProjectRequest request) throws FailedToCreateProjectException {
+        try {
+            int id = projectDao.createProject(request);
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/api/ProjectService.java]: should you validate the request object?

File: src/main/java/org/kainos/ea/api/ProjectService.java [hunk 2]
```diff
@@ -13,6 +16,10 @@ public ProjectService(ProjectDao projectDao) {
         this.projectDao = projectDao;
     }
 
+    public List<Project> getProjects() throws FailedToGetProjectException {
+        return projectDao.getProjects();
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/api/ProjectService.java]: catch sql exception and throw customer exception

File: src/main/java/org/kainos/ea/api/ProjectService.java [hunk 3]
```diff
@@ -21,4 +28,36 @@ public Project getProjectById(int id) throws FailedToGetProjectException, Projec
...
+        }
+    }
+
+    public void updateProject(int id, ProjectRequest request) throws FailedToUpdateProjectException, ProjectDoesNotExistException {
+        try {
+            if (projectDao.getProjectById(id) == null) {
+                throw new ProjectDoesNotExistException();
+            }
+
+             projectDao.updateProject(id, request);
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/api/ProjectService.java:75]: should you validate this request?

File: src/main/java/org/kainos/ea/db/ProjectDao.java [hunk 4]
```diff
@@ -1,15 +1,47 @@
...
-import java.sql.SQLException;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
 
 public class ProjectDao {
 
+    public List<Project> getProjects() throws FailedToGetProjectException {
+
+        String queryString = "SELECT * FROM Projects";
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/db/ProjectDao.java]: avoid using *

**Review by shaunganley — CHANGES_REQUESTED** [review]

**Review by shaunganley — APPROVED** [review]

**Review by Adaray786 — APPROVED** [review]

**Pull request merged**